src/thread/thread_worker_pool.hpp

```diff
@@ -1013,26 +1047,26 @@ namespace cubthread
   void
   worker_pool<Context>::core::worker::push_task_on_new_thread (task<Context> *work_p, cubperf::time_point push_time)
   {
-    // start new thread and run task
-    assert (work_p != NULL);
-
-    // make sure worker is in a valid state
-    assert (m_task_p == NULL);
-    assert (m_context_p == NULL);
-
-    // save push time
-    m_push_time = push_time;
-
-    // save task
-    m_task_p = work_p;
-
-    // start thread.
-    //
-    // the next code tries to help visualizing any system errors that can occur during create or detach in debug mode
-    //
-    // release will basically be reduced to:
-    // std::thread (&worker::run, this).detach ();
-    //
+    if (work_p != NULL)
```

Comment:
Remove if condition and `assert (m_task_p == NULL);` safe-guard. Other operations and comments are not restricted to `m_task_p != NULL` case.